From 827c5a6b24d18e5b5089628087066a28a52fbaf4 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Tue, 14 Nov 2023 13:06:20 +0200
Subject: [PATCH] [SW-15774] build: consolidate common defines

Create a common file under the zephyr package for holding the common
configuration.

Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
Change-Id: I44d299b1f3e53b272366f59aec03300b69d56279
---

diff --git a/package/agent-fwl/agent-fwl.mk b/package/agent-fwl/agent-fwl.mk
index 45be4a4..921b44a 100644
--- a/package/agent-fwl/agent-fwl.mk
+++ b/package/agent-fwl/agent-fwl.mk
@@ -14,9 +14,9 @@
 
 AGENT_FWL_CONF_APP_EXTRA_CFLAGS = \
 		-include $(EMBEDDED_ROOT)/specs/$(AGENT_FWL_DEVICE_NAME)/$(AGENT_FWL_DEVICE_NAME)_boot_info.h \
-		$($(call UPPERCASE,$(DEV_PREFIX))_CONF_DEV_EXTRA_CFLAGS) \
+		$(CONF_DEV_EXTRA_CFLAGS) \
+		$(COMMON_CONF_EXTRA_CFLAGS) \
 		$(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)
-
 
 AGENT_FWL_CONF_APP_CFLAGS = \
 		-I$(AGENT_FWL_SRC_DIR)/../common/include \
@@ -46,13 +46,14 @@
 AGENT_FWL_CONF_OPTS += -DEXTRA_CFLAGS="\
 		-Werror \
 		-Wno-misleading-indentation \
-		$($(call UPPERCASE,$(DEV_PREFIX))_CONF_DEV_EXTRA_CFLAGS) \
+		$(CONF_DEV_EXTRA_CFLAGS) \
 		$(AGENT_FWL_CONF_APP_EXTRA_CFLAGS)"
 
 AGENT_FWL_CONF_OPTS += -DFAKE_SECURITY=$(FAKE_SECURITY)
 AGENT_FWL_CONF_OPTS += -DBUILD_VERSION="$(AGENT_FWL_APP_VERSION)"
 AGENT_FWL_CONF_OPTS += -DBUILD_MACHINE="$(shell hostname -s)"
 AGENT_FWL_CONF_OPTS += -DCMAKE_C_FLAGS="\
+		$(CONF_DEV_EXTRA_CFLAGS) \
 		$(AGENT_FWL_CONF_APP_CFLAGS)"
 
 AGENT_FWL_CONF_OPTS += -DDTS_EXTRA_CPPFLAGS="-I$(SPECS_EMBEDDED_DIR)"
diff --git a/package/agent-hbm/agent-hbm.mk b/package/agent-hbm/agent-hbm.mk
index 4918c3a..3455a47 100644
--- a/package/agent-hbm/agent-hbm.mk
+++ b/package/agent-hbm/agent-hbm.mk
@@ -13,7 +13,7 @@
 AGENT_HBM_LOCAL_VERSION = $$(sed "s/^.*fw-\\([0-9]\\+\\.[0-9]\\+\\.[0-9]\\+\\).*$$/\\1/" $(@D)/localversion-habanalabs)
 
 AGENT_HBM_CONF_APP_EXTRA_CFLAGS = \
-		-include $(EMBEDDED_ROOT)/specs/$(AGENT_HBM_DEVICE_NAME)/$(AGENT_HBM_DEVICE_NAME)_boot_info.h \
+		$(COMMON_CONF_EXTRA_CFLAGS) \
 		$(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)
 
 AGENT_HBM_CONF_APP_CFLAGS = \
@@ -51,11 +51,8 @@
 AGENT_HBM_CONF_OPTS += -DBUILD_VERSION="$(AGENT_HBM_APP_VERSION)"
 AGENT_HBM_CONF_OPTS += -DBUILD_MACHINE="$(shell hostname -s)"
 AGENT_HBM_CONF_OPTS += -DCMAKE_C_FLAGS="\
-		-I$(SPECS_EMBEDDED_DIR) \
-		-I$(SPECS_EMBEDDED_DIR)/$(DEV_PREFIX) \
-		-I$(SPECS_ROOT) \
-		-I$(SPECS_EXT_ROOT) \
-		-I$(SPECS_ROOT)/$(DEV_PREFIX)/asic_reg \
+		$(CONF_DEV_EXTRA_CFLAGS) \
+		$(COMMON_SPECS_CFLAGS) \
 		$(AGENT_HBM_CONF_APP_CFLAGS)"
 
 AGENT_HBM_CONF_OPTS += -DDTS_EXTRA_CPPFLAGS="-I$(SPECS_EMBEDDED_DIR)"
diff --git a/package/agent-pcie/agent-pcie.mk b/package/agent-pcie/agent-pcie.mk
index 4b1e033..186685f 100644
--- a/package/agent-pcie/agent-pcie.mk
+++ b/package/agent-pcie/agent-pcie.mk
@@ -13,7 +13,7 @@
 AGENT_PCIE_LOCAL_VERSION = $$(sed "s/^.*fw-\\([0-9]\\+\\.[0-9]\\+\\.[0-9]\\+\\).*$$/\\1/" $(@D)/localversion-habanalabs)
 
 AGENT_PCIE_CONF_APP_EXTRA_CFLAGS = \
-		-include $(EMBEDDED_ROOT)/specs/$(AGENT_PCIE_DEVICE_NAME)/$(AGENT_PCIE_DEVICE_NAME)_boot_info.h \
+		$(COMMON_CONF_EXTRA_CFLAGS) \
 		$(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)
 
 AGENT_PCIE_CONF_APP_CFLAGS = \
@@ -44,11 +44,7 @@
 AGENT_PCIE_CONF_OPTS += -DBUILD_VERSION="$(AGENT_PCIE_APP_VERSION)"
 AGENT_PCIE_CONF_OPTS += -DBUILD_MACHINE="$(shell hostname -s)"
 AGENT_PCIE_CONF_OPTS += -DCMAKE_C_FLAGS="\
-		-I$(SPECS_EMBEDDED_DIR) \
-		-I$(SPECS_EMBEDDED_DIR)/$(DEV_PREFIX) \
-		-I$(SPECS_ROOT) \
-		-I$(SPECS_EXT_ROOT) \
-		-I$(SPECS_ROOT)/$(DEV_PREFIX)/asic_reg \
+		$(COMMON_SPECS_CFLAGS) \
 		$(AGENT_PCIE_CONF_APP_CFLAGS)"
 
 AGENT_PCIE_CONF_OPTS += -DDTS_EXTRA_CPPFLAGS="-I$(SPECS_EMBEDDED_DIR)"
diff --git a/package/agent-serdes/agent-serdes.mk b/package/agent-serdes/agent-serdes.mk
index 69d0fd0..8b46719 100644
--- a/package/agent-serdes/agent-serdes.mk
+++ b/package/agent-serdes/agent-serdes.mk
@@ -13,8 +13,7 @@
 AGENT_SERDES_LOCAL_VERSION = $$(sed "s/^.*fw-\\([0-9]\\+\\.[0-9]\\+\\.[0-9]\\+\\).*$$/\\1/" $(@D)/localversion-habanalabs)
 
 AGENT_SERDES_CONF_APP_EXTRA_CFLAGS = \
-		-include $(EMBEDDED_ROOT)/specs/$(AGENT_SERDES_DEVICE_NAME)/$(AGENT_SERDES_DEVICE_NAME)_boot_info.h \
-		$($(call UPPERCASE,$(DEV_PREFIX))_CONF_DEV_EXTRA_CFLAGS) \
+		$(COMMON_CONF_EXTRA_CFLAGS) \
 		$(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)
 
 AGENT_SERDES_CONF_APP_CFLAGS = \
diff --git a/package/event/event.mk b/package/event/event.mk
index a3332ed..6ca1760 100644
--- a/package/event/event.mk
+++ b/package/event/event.mk
@@ -13,21 +13,13 @@
 EVENT_ZEPHYR_BUILD_DIR ?= $(@D)/zephyr
 
 EVENT_CONF_CFLAGS =\
-		-I$(EVENT_ZEPHYR_BUILD_DIR) \
-		-I$(EVENT_ZEPHYR_BUILD_DIR)/include \
-		-I$(EVENT_ZEPHYR_BUILD_DIR)/include/drivers/misc/hl_npu \
-		-I$(EVENT_ZEPHYR_BUILD_DIR)/include/drivers/misc/hl_npu/$(HL_DEV) \
-		-I$(EVENT_ZEPHYR_BUILD_DIR)/include/drivers/misc/hl_npu/common \
-		-I$(EVENT_ZEPHYR_BUILD_DIR)/drivers/misc/hl_npu \
-		-I$(EVENT_ZEPHYR_BUILD_DIR)/drivers/misc/hl_npu/$(HL_DEV) \
-		-I$(EVENT_ZEPHYR_BUILD_DIR)/drivers/misc/hl_npu/common \
-		-I$(EVENT_ZEPHYR_BUILD_DIR)/include/generated \
-		-I$(EVENT_ZEPHYR_BUILD_DIR)/include/generated/syscalls \
+		$(COMMON_DRV_CONF_CFLAGS) \
 		$(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)
 
 EVENT_CONF_EXTRA_CFLAGS = \
 		-include $(EMBEDDED_ROOT)/specs/$(EVENT_DEVICE_NAME)/$(EVENT_DEVICE_NAME)_boot_info.h \
 		$(CONF_DEV_EXTRA_CFLAGS) \
+		$(COMMON_CONF_EXTRA_CFLAGS) \
 		$(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)
 
 EVENT_CONF_ENV = CROSS_COMPILE=$(TARGET_CROSS)
diff --git a/package/mgmt/mgmt.mk b/package/mgmt/mgmt.mk
index 114c17c..970d15c 100644
--- a/package/mgmt/mgmt.mk
+++ b/package/mgmt/mgmt.mk
@@ -13,21 +13,13 @@
 MGMT_ZEPHYR_BUILD_DIR ?= $(@D)/zephyr
 
 MGMT_CONF_CFLAGS =\
-		-I$(MGMT_ZEPHYR_BUILD_DIR) \
-		-I$(MGMT_ZEPHYR_BUILD_DIR)/include \
-		-I$(MGMT_ZEPHYR_BUILD_DIR)/include/drivers/misc/hl_npu \
-		-I$(MGMT_ZEPHYR_BUILD_DIR)/include/drivers/misc/hl_npu/$(HL_DEV) \
-		-I$(MGMT_ZEPHYR_BUILD_DIR)/include/drivers/misc/hl_npu/common \
-		-I$(MGMT_ZEPHYR_BUILD_DIR)/drivers/misc/hl_npu \
-		-I$(MGMT_ZEPHYR_BUILD_DIR)/drivers/misc/hl_npu/$(HL_DEV) \
-		-I$(MGMT_ZEPHYR_BUILD_DIR)/drivers/misc/hl_npu/common \
-		-I$(MGMT_ZEPHYR_BUILD_DIR)/include/generated \
-		-I$(MGMT_ZEPHYR_BUILD_DIR)/include/generated/syscalls \
+		$(COMMON_DRV_CONF_CFLAGS) \
 		$(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)
 
 MGMT_CONF_EXTRA_CFLAGS =\
 		-include $(EMBEDDED_ROOT)/specs/$(MGMT_DEVICE_NAME)/$(MGMT_DEVICE_NAME)_boot_info.h \
 		$(CONF_DEV_EXTRA_CFLAGS) \
+		$(COMMON_CONF_EXTRA_CFLAGS) \
 		$(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)
 
 MGMT_CONF_ENV = CROSS_COMPILE=$(TARGET_CROSS)
